### ISSUE TITLE
[Grid - Grouping] Accordions missing expanded/collapsed state for assistive technologies

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2883]
### Provide a general summary of the issue here

The accordion buttons in the grouped variant are missing aria-expanded attribute. Assistive technology users (especially screen reader users) must explore the surrounding elements to understand whether the element is closed or opened. 

<img width="1410" height="239" alt="Image" src="https://github.com/user-attachments/assets/14b482ab-0c18-4331-bf62-ee833e5404ed" />

### 🤔 Expected Behavior?

When user operates the accordion button users would head "expanded" or "collapsed" (or similar depending on screen reader)

### 😯 Current Behavior

When user operates the accordion button screen readers stay silent (some might repeat the current button)

### 💁 Possible Solution

Add aria-expanded with value "true" when the accordion is open and "false" when it is closed. 

### 🖥️ Steps to Reproduce

1. Navigate to accordion button with a screen reader
2. Activate it (Enter or Space key) -> Screen reader stays silent

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
